nislmigrate/migrators/file_migrator.py

```diff
@@ -118,19 +118,25 @@ def restore(self, migration_directory: str, facade_factory: FacadeFactory, argum
             configuration.mongo_configuration,
             migration_directory,
             self.name)
-
-        if configuration.should_update_store:
-            self.update_root_file_path_in_metadata(configuration)
-
-        if configuration.use_forward_slashes:
-            self.update_file_path_slashes_in_metadata(configuration)
-
+        self.modify(self, migration_directory, facade_factory, arguments)
         if configuration.should_migrate_files:
             configuration.file_facade.copy_directory(
                 configuration.file_migration_directory,
                 configuration.data_directory,
                 True)
 
+    def modify(self, migration_directory: str, facade_factory: FacadeFactory, arguments: Dict[str, Any]):
+        configuration = _FileMigratorConfiguration(
+            migration_directory,
+            facade_factory,
+            arguments,
+            self.config(facade_factory)
+        )
+        if configuration.should_update_store:
```

Comment:
This won't work. The update code relies on a file existing on disk containing the old path information. You will need to provide that information as an argument for this operation.

Reply:
I've added a new argument `--files-file-store-root` and used it when it in the modify code path and used the file in the restore code path. I also added a pre check that if you're running `modify --files --files-change-file-store-root` you must also pass `--files-file-store-root`.

This does highlight that we're showing arguments that don't make sense for modify when running the help. To improve this we'd need to know whether a migrator supports `migrate` and chop up the argument parser accordingly. Here's an example of the help:
```
❯ poetry run nislmigrate modify --help
usage: nislmigrate modify [-h] [-d] [-s] [--dir DIR] [--all] [--alarms]       
                          [--assets] [--assetrule] [--dashboards] [--files]   
                          [--files-metadata-only]
                          [--files-change-file-store-root new-root-dir]       
                          [--files-file-store-root existing-root-dir]
                          [--files-switch-to-forward-slashes] [--notification]
                          [--repo] [--security] [--systemstates] [--systems]  
                          [--tags] [--tagrule] [--testmonitor] [--userdata]   
                          [--secret <secret>]

optional arguments:
  -h, --help            show this help message and exit
  -d, --debug           print all logged information and stack trace
                        information in case an error occurs
  -s, --silent          print all logged information except debugging
                        information
  --dir DIR             specify the directory used for migrated data (defaults
                        to documents)
  --all                 use all provided migrator plugins during a capture or 
                        restore operation
  --alarms              Migrate alarm instances
  --assets              Migrate asset utilization and calibration data
  --assetrule           Migrate asset management alarm rules
  --dashboards          Migrate dashboards and web applications
  --files               Migrate ingested files
  --files-metadata-only
                        When used with --files or --all, migrate only the file
                        metadata, but not the files themselves. Otherwise
                        ignored.
  --files-change-file-store-root new-root-dir
                        Change the file storage location path
  --files-file-store-root existing-root-dir
                        The file store root directory. Required when "--files-
                        change-file-store-root" is specified with the "modify"
                        operation, ignored otherwise.
  --files-switch-to-forward-slashes
                        Change the file storage location path from backward
                        slashes to forward slashes
  --notification        Migrate notifications strategies, templates, and
                        groups
  --repo                Migrate packages and feeds
  --security            Migrate workspaces.
  --systemstates        Migrate system states
  --systems             Migrate registered systems. Must include the --secret
                        <SECRET> command line argument when using this
                        migrator.
  --tags                migrate tags and tag histories
  --tagrule             Migrate Tag alarm rules
  --testmonitor         Migrate notifications strategies, templates, and
                        groups
  --userdata            Migrate user data
  --secret <secret>     Some migrators require this --secret to encrypt
                        sensitive data during migration otherwise it is
                        ignored. You will need to provide the same password
                        when restoring and capturing data.
```

Reply:
Should we create an issue to improve the arguments list for modify?

Reply:
Created Issue #144